main/src/ca/uwaterloo/flix/MutationTester.scala

```diff
@@ -0,0 +1,86 @@
+package ca.uwaterloo.flix
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.ast.Ast.Constant
+import ca.uwaterloo.flix.language.ast.{Ast, TypedAst}
+import ca.uwaterloo.flix.language.ast.TypedAst.Expr
+import ca.uwaterloo.flix.util._
+
+import java.io.File
+
+object MutationTester {
+    def run(files: Seq[File], options: Options, tester: String, testee: String): Unit = {
+        val flix = new Flix
+        flix.setOptions(options)
+        flix.addSourceCode("testee",
+            s"""
+            |def main(): Unit \\ IO =
+            |    println(Test.one())
+            |
+            |mod Test {
+            |    pub def one(): Int32 = 1
+            |}""".stripMargin)
+
+        flix.addSourceCode("tester",
+            s"""
+            |mod MainTest {
+            |    @test
+            |    def testConstant(): Bool =
+            |        Assert.eq(Test.one(), 1)
+            |}""".stripMargin)
+        val root = flix.check().unsafeGet
+        val root1 = mutate(root, testee)
+        val res = flix.codeGen(root1).unsafeGet.getTests
+        println(res)
+        /**
+        val result = root1.map(r => flix.codeGen(r).unsafeGet)
+        val tests = result.map(res => res.getTests)
+        */
+    }
+
+    def mutate(root: TypedAst.Root, testee: String): TypedAst.Root = {
+        val defs = root.defs
+        //defs.toList.map(t => mutateExpr(t._2.exp))
+        val defSyms = root.modules.filter(pair => (pair._1.toString.equals(testee))).values.toList.flatten
+        println(defSyms)
+        val newDefs = defs.map(d => (d._1,d._2) match {
+            case (s, fun) =>
+                if (defSyms.contains(s)) {
+                    println(d._2)
+                    val mut = TypedAst.Def(fun.sym, fun.spec, mutateExpr(fun.exp))
+                    println(mut)
+                    d._1 -> mut
+
+                } else d
+            case _ => d
+        })
+        root.copy(defs = newDefs)
```

Comment:
The above code is of course horrendous. I would suggest: (a) introduce a `visitDef` and (b) pick _one_ def to mutate.

Remember you are not doing higher-order mutation testing! You are only supposed  to make _one_ change.